### ISSUE TITLE
XGBoost.to_json

### ISSUE THREAD

**Issue by oualib** [vertica/VerticaPy#169]
The idea of this function is to convert any Vertica XGBoost Model to the open sources XGBoost API. This API supports a specific JSON format that could be generated.

There are points to know in order to understand the conversion:
 - The JSON file is case sensitive. For example, list of boolean values must be in lower case.
 - List of numerical values can not include integers. For example, the following list: [1, 2.5, 4.5, 2, 8.5] is incorrect whereas [1.0, 2.5, 4.5, 2.0, 8.5] is correct.
 - split_conditions list must include the Split Condition for the non-leaf nodes but the predictions for the leaves.
 - loss_changes, base_weights and sum_hessian can not be generated by Vertica as these information are not stored. Instead, we are generating a list with only zeros (0.0).
 - For regression and binary classification, base_score corresponds to the average of the target. It is used during the scoring. For these 2 types, the matching will be very precise (both prediction and probability).
 - For multi class classifications, a tree is stored for each class. n trees in Vertica will lead to n * nc trees in the XGBoost API (nc being the number of classes)
 -  For multi class classifications, base_score is useless and set to 5E-1.
 - For multi class classification, the prediction will be the same (also the sorted arguments). However, the probability will differ. Vertica is using multi-logistic function to normalize the data whereas the XGBoost API is using Softmax.
 - The sklearn call string is not important in case of regression and binary classification but it is needed for multi class classification. Otherwise, the returned results are weird... They may be some intermediate results computed by the Python library.
 - Some list are empty but must be in the file otherwise the parsing will not work.
 - Double quotes can not be replaced by single ones.
